cmd/worker/internal/sourcegraphaccounts/notifications_subscriber.go

```diff
@@ -0,0 +1,235 @@
+package sourcegraphaccounts
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/sourcegraph/log"
+	sams "github.com/sourcegraph/sourcegraph-accounts-sdk-go"
+	clientsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1"
+	notificationsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/notifications/v1"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+	"golang.org/x/oauth2/google"
+
+	"github.com/sourcegraph/sourcegraph/cmd/worker/job"
+	workerdb "github.com/sourcegraph/sourcegraph/cmd/worker/shared/init/db"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/dotcom"
+	"github.com/sourcegraph/sourcegraph/internal/env"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+var _ job.Job = (*notificationsSubscriber)(nil)
+
+// notificationsSubscriber is a worker responsible for receiving notifications
+// from Sourcegraph Accounts.
+type notificationsSubscriber struct {
+	config *notificationsSubscriberConfig
+}
+
+func NewNotificationsSubscriber() job.Job {
+	return &notificationsSubscriber{
+		config: &notificationsSubscriberConfig{},
+	}
+}
+
+func (s *notificationsSubscriber) Description() string {
+	return "Receives notifications from Sourcegraph Accounts."
+}
+
+func (s *notificationsSubscriber) Config() []env.Config {
+	return []env.Config{s.config}
+}
+
+func (s *notificationsSubscriber) Routines(ctx context.Context, observationCtx *observation.Context) ([]goroutine.BackgroundRoutine, error) {
+	logger := observationCtx.Logger.Scoped("sourcegraphAccounts.notificationsSubscriber")
+	if !dotcom.SourcegraphDotComMode() || s.config.GCP.CredentialsFile == "" {
+		logger.Debug("worker disabled because it is not running on Sourcegraph.com or missing GCP credentials file")
+		return nil, nil
+	}
```

Comment:
I'm not sure if we have a convention or not, but this is a log that will show up in customer instances as well. IMO we should silently exit outside dotcom, and fail/succeed with more visible logs in dotcom:

```suggestion
	if !dotcom.SourcegraphDotComMode() {
		return nil, nil // not relevant
	}
	if s.config.GCP.CredentialsFile == "" {
		logger.Info("worker disabled because SOURCEGRAPH_ACCOUNTS_CREDENTIALS_FILE GCP is not set")
		return nil, nil
	}
	logger.Debug("worker enabled")
```

Reply:
Good call, updated in https://github.com/sourcegraph/sourcegraph/pull/63051/commits/42cf67cadd042b9fb62eb27e01b707401c5e0f5f